src/main/groovy/com/jetbrains/python/envs/PythonEnvsExtension.groovy

```diff
@@ -7,85 +7,215 @@ package com.jetbrains.python.envs
      * I.e. "python" will be have "python2.7" link (same for exe file). Used for envs line tox.
      */
-    void conda(final String envName, final String version, final List<String> packages, final boolean linkWithVersion) {
-        def pipPackages = packages.findAll { !it.startsWith(CONDA_PREFIX) }
-        def condaPackages = packages.findAll { it.startsWith(CONDA_PREFIX) }.collect {
-            it.substring(CONDA_PREFIX.length())
+    void conda(final String envName,
+               final String version,
+               final String architecture = null,
+               final List<String> packages = null,
+               final boolean linkWithVersion = false) {
+        List<String> pipPackages = packages.findAll { !it.startsWith(CONDA_PREFIX) }
+        List<String> condaPackages = packages.findAll { it.startsWith(CONDA_PREFIX) }
+                                             .collect { it.substring(CONDA_PREFIX.length()) }
+        condaEnvs << new CondaEnv(envName, envsDirectory, version, is64(architecture), pipPackages, condaPackages, linkWithVersion)
+    }
+
+    /**
+     * @see #python
+     */
+    void jython(final String envName, final List<String> packages = null) {
+        pythonEnvs << new PythonEnv(envName, envsDirectory, EnvType.JYTHON, null, null, packages)
+    }
+
+    /**
+     * @see #python
+     */
+    void pypy(final String envName, final List<String> packages = null, final String version = null) {
+        pythonEnvs << new PythonEnv(
+                envName,
+                envsDirectory,
+                EnvType.PYPY,
+                version ?: "pypy2.7-5.8.0",
+                null,
+                packages
+        )
+    }
+
+    /**
+     * @see #python
+     */
+    void ironpython(final String envName,
+                    final List<String> packages = null,
+                    final String architecture = null,
+                    final URL urlToArchive = null) {
+        URL urlToIronPythonZip = new URL("https://github.com/IronLanguages/main/releases/download/ipy-2.7.7/IronPython-2.7.7-win.zip")
+        envsFromZip << new PythonEnv(
+                envName,
+                envsDirectory,
+                EnvType.IRONPYTHON,
+                null,
+                is64(architecture),
+                packages,
+                urlToArchive ?: urlToIronPythonZip
+        )
+    }
+
+    /**
+     * @see #python
+     * @param sourceEnvName name of inherited environment like "env_for_django"
+     */
+    void virtualenv(final String envName, final String sourceEnvName, final List<String> packages = null) {
+        PythonEnv pythonEnv = allEnvs().find { it.name == sourceEnvName }
+        if (pythonEnv != null) {
+            virtualEnvs << new VirtualEnv(envName, virtualenvsDirectory ?: envsDirectory, pythonEnv, packages)
+        } else {
+            println("\u001B[31mSpecified environment '$sourceEnvName' for virtualenv '$envName' isn't found\u001B[0m")
```

Comment:
Escape sequences make messages better in the console, but on the CI it can make it less readable.